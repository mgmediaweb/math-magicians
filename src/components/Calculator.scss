.calculatorBody {
    background-color: white;
    height: auto;
    margin: 0;
    max-width: 480px;
    padding: 10px;
    width: 100%;
}

.rowBtn {
    display: grid;
}

.normal {
    grid-template-columns: 1fr 1fr 1fr 1fr;
}

.double {
    grid-template-columns: 2fr 1fr 1fr;
}