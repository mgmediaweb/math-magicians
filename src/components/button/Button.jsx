import PropTypes from 'prop-types';
import './Button.scss';

function Button({ text, type, axn }) {
  const clickAxn = (opc) => axn(opc);

  return (
    <div
      className={`btn btn-${type}`}
      role="button"
      onClick={() => clickAxn(text)}
      onKeyUp={() => clickAxn(text)}
      tabIndex={text}
    >
      {text}
    </div>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  axn: PropTypes.func.isRequired,
};

export default Button;
