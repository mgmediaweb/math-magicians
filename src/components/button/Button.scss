$color-btn-normal: #e0e0e0;
$color-btn-operator: #f39148;

$border-width: 1px;

.btn {
  align-items: center;
  background-color: $color-btn-normal;
  box-sizing: border-box;
  color: black;
  display: flex;
  height: 66px;
  font-size: 1.1rem;
  justify-content: center;
  margin: 0;
  padding: 0;
  width: 100%;
  user-select: none;
}

.double {
  grid-column: 1/3;
}

.btn-normal {
  background-color: $color-btn-normal;
  border: $border-width solid darken($color-btn-normal, 10%);

  &:hover {
    cursor: pointer;
    background-color: darken($color-btn-normal, 5%);
  }
}

.btn-orange {
  background-color: $color-btn-operator;
  border: $border-width solid darken($color-btn-operator, 10%);

  &:hover {
    cursor: pointer;
    background-color: darken($color-btn-operator, 5%);
  }
}
