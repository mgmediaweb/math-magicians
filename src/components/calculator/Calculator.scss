.calculatorBody {
  background-color: white;
  display: block;
  height: auto;
  margin: 0;
  max-width: 480px;
  padding: 10px;
  width: 100%;
}

.rowBtns {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: auto;
}
