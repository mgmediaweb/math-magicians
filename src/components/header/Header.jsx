import calcImage from '../../assets/images/calculator.png';
import './header.scss';

const Header = () => (
  <div className="header">
    <div className="container d-flex">
      <div className="logo">
        <img src={calcImage} alt="Math Magicians" />
        <h1>Math Magicians</h1>
      </div>
      <nav>
        <ul>
          <li><a href="/">Home</a></li>
          <li><a href="/calculator">Calculator</a></li>
          <li><a href="/quote">Quote</a></li>
        </ul>
      </nav>
    </div>
  </div>
);

export default Header;
