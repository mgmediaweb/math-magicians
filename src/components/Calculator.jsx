import React from 'react';
import Button from './button/Button';
import Screen from './screen/Screen';
import './Calculator.scss';

class Calculator extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      screen: '0',
    };

    this.axn = this.axn.bind(this);
  }

  axn(opc) {
    this.setState((state) => {
      if (opc === '=') {
        console.log('go result');
      } else if (opc === 'AC') {
        return { screen: '0' };
      } else {
        let ans = state.screen + opc;

        if (state.screen === '0') {
          ans = opc;
        }

        return { screen: ans };
      }

      return true;
    });
  }

  render() {
    const { screen } = this.state;
    return (
      <div className="calculatorBody">
        <Screen info={screen} />
        <div className="rowBtn normal">
          <Button text="AC" type="normal" axn={this.axn} />
          <Button text="+/-" type="normal" axn={this.axn} />
          <Button text="%" type="normal" axn={this.axn} />
          <Button text="/" type="operator" axn={this.axn} />
        </div>
        <div className="rowBtn normal">
          <Button text="7" type="normal" axn={this.axn} />
          <Button text="8" type="normal" axn={this.axn} />
          <Button text="9" type="normal" axn={this.axn} />
          <Button text="X" type="operator" axn={this.axn} />
        </div>
        <div className="rowBtn normal">
          <Button text="4" type="normal" axn={this.axn} />
          <Button text="5" type="normal" axn={this.axn} />
          <Button text="6" type="normal" axn={this.axn} />
          <Button text="-" type="operator" axn={this.axn} />
        </div>
        <div className="rowBtn normal">
          <Button text="1" type="normal" axn={this.axn} />
          <Button text="2" type="normal" axn={this.axn} />
          <Button text="3" type="normal" axn={this.axn} />
          <Button text="+" type="operator" axn={this.axn} />
        </div>
        <div className="rowBtn double">
          <Button text="0" type="normal" axn={this.axn} />
          <Button text="." type="normal" axn={this.axn} />
          <Button text="=" type="operator" axn={this.axn} />
        </div>
      </div>
    );
  }
}

export default Calculator;
