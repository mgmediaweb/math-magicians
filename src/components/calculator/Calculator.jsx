import React, { useState } from 'react';
import Button from '../button/Button';
import Screen from '../screen/Screen';
import Calculate from '../../logic/calculate';
import './Calculator.scss';

function Calculator() {
  const [state, setState] = useState({ total: null, next: null, operation: null });

  const setDisplay = (content) => {
    setState(() => ({ next: content }));
  };

  const axn = (opc) => {
    setState(Calculate(state, opc));
  };

  const buttons = [
    { text: 'AC', type: 'normal' },
    { text: '+/-', type: 'normal' },
    { text: '%', type: 'normal' },
    { text: 'รท', type: 'orange' },
    { text: '7', type: 'normal' },
    { text: '8', type: 'normal' },
    { text: '9', type: 'normal' },
    { text: 'x', type: 'orange' },
    { text: '4', type: 'normal' },
    { text: '5', type: 'normal' },
    { text: '6', type: 'normal' },
    { text: '-', type: 'orange' },
    { text: '1', type: 'normal' },
    { text: '2', type: 'normal' },
    { text: '3', type: 'normal' },
    { text: '+', type: 'orange' },
    { text: '0', type: 'normal double' },
    { text: '.', type: 'normal' },
    { text: '=', type: 'orange' },
  ];

  return (
    <div className="calculatorBody">
      <Screen info={state.next || state.total || '0'} onChange={(e) => setDisplay(e.target.value)} />
      <div className="rowBtns">
        {
          buttons.map((btn) => (
            <Button
              key={btn.text}
              text={btn.text}
              type={btn.type}
              axn={axn}
            />
          ))
        }
      </div>
    </div>
  );
}

export default Calculator;
