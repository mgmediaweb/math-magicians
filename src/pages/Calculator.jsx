import Calculator from '../components/calculator/Calculator';
import './pages.scss';

const CalculatorScreen = () => (
  <div className="screen">
    <div className="container twoColumns">
      <div>
        <h2>Let&apos;s do some Math!</h2>
      </div>
      <Calculator />
    </div>
  </div>
);

export default CalculatorScreen;
