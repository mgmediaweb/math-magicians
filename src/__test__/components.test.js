import React from 'react';
import TestRenderer from 'react-test-renderer';
import Header from '../components/header/Header';
import HomeScreen from '../pages/Home';
import CalculatorScreen from '../pages/Calculator';
import QuoteScreen from '../pages/Quote';
import Calculator from '../components/calculator/Calculator';
import Button from '../components/button/Button';

describe('testing react components using jest snapshot', () => {

  it('Testing Header component', () => {
    const testRenderer = TestRenderer.create(<Header />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

  it('Testing Home component', () => {
    const testRenderer = TestRenderer.create(<HomeScreen />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

  it('Testing Calculator component', () => {
    const testRenderer = TestRenderer.create(<CalculatorScreen />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });    

  it('Testing Quote component', () => {
    const testRenderer = TestRenderer.create(<QuoteScreen />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

  it('Testing Calculator component', () => {
    const testRenderer = TestRenderer.create(<Calculator />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });  

  it('Testing Button component', () => {
    const testRenderer = TestRenderer.create(<Button
      key="+"
      text="+"
      type="orange"
      axn={() => { console.log('1'); }}
    />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});