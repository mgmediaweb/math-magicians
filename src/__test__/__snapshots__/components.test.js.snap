// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing react components using jest snapshot Testing Button component 1`] = `
<div
  className="btn btn-orange"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex="+"
>
  +
</div>
`;

exports[`testing react components using jest snapshot Testing Calculator component 1`] = `
<div
  className="screen"
>
  <div
    className="container twoColumns"
  >
    <div>
      <h2>
        Let's do some Math!
      </h2>
    </div>
    <div
      className="calculatorBody"
    >
      <div
        className="screenCalc"
        data-testid="mainScreen"
      >
        0
      </div>
      <div
        className="rowBtns"
      >
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="AC"
        >
          AC
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="+/-"
        >
          +/-
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="%"
        >
          %
        </div>
        <div
          className="btn btn-orange"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="รท"
        >
          รท
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="7"
        >
          7
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="8"
        >
          8
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="9"
        >
          9
        </div>
        <div
          className="btn btn-orange"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="x"
        >
          x
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="4"
        >
          4
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="5"
        >
          5
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="6"
        >
          6
        </div>
        <div
          className="btn btn-orange"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="-"
        >
          -
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="1"
        >
          1
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="2"
        >
          2
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="3"
        >
          3
        </div>
        <div
          className="btn btn-orange"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="+"
        >
          +
        </div>
        <div
          className="btn btn-normal double"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="0"
        >
          0
        </div>
        <div
          className="btn btn-normal"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="."
        >
          .
        </div>
        <div
          className="btn btn-orange"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex="="
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing react components using jest snapshot Testing Calculator component 2`] = `
<div
  className="calculatorBody"
>
  <div
    className="screenCalc"
    data-testid="mainScreen"
  >
    0
  </div>
  <div
    className="rowBtns"
  >
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="AC"
    >
      AC
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="+/-"
    >
      +/-
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="%"
    >
      %
    </div>
    <div
      className="btn btn-orange"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="รท"
    >
      รท
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="7"
    >
      7
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="8"
    >
      8
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="9"
    >
      9
    </div>
    <div
      className="btn btn-orange"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="x"
    >
      x
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="4"
    >
      4
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="5"
    >
      5
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="6"
    >
      6
    </div>
    <div
      className="btn btn-orange"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="-"
    >
      -
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="1"
    >
      1
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="2"
    >
      2
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="3"
    >
      3
    </div>
    <div
      className="btn btn-orange"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="+"
    >
      +
    </div>
    <div
      className="btn btn-normal double"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      0
    </div>
    <div
      className="btn btn-normal"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="."
    >
      .
    </div>
    <div
      className="btn btn-orange"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="="
    >
      =
    </div>
  </div>
</div>
`;

exports[`testing react components using jest snapshot Testing Header component 1`] = `
<div
  className="header"
>
  <div
    className="container d-flex"
  >
    <div
      className="logo"
    >
      <img
        alt="Math Magicians"
        src="calculator.png"
      />
      <h1>
        Math Magicians
      </h1>
    </div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`testing react components using jest snapshot Testing Home component 1`] = `
<div
  className="screen"
>
  <div
    className="container"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum vitae quam vel neque pulvinar tincidunt. Nulla consectetur, ante ut viverra fringilla, est urna ultricies mauris, ac fringilla risus neque id lectus. Phasellus porta enim a nibh tincidunt, id sollicitudin libero egestas. Praesent vitae massa scelerisque turpis feugiat faucibus id a dui. Duis laoreet vehicula elit, eu eleifend lorem iaculis eu. Aenean non ante id velit lobortis faucibus. Donec pretium convallis nisl. Vestibulum viverra efficitur libero et suscipit. Donec non nulla id enim imperdiet interdum. Nullam viverra magna non mi venenatis tincidunt. Nam vitae lacinia nisl. Etiam quis neque justo. Morbi consectetur diam et cursus cursus. Cras rhoncus semper nisi, convallis molestie tortor. Nulla scelerisque lorem augue, et pharetra diam malesuada id.
    </p>
    <p>
      Proin vitae imperdiet risus. Etiam id lacinia risus. Suspendisse lobortis felis turpis, non   sodales lacus consequat eget. Nam quis sem a lectus tincidunt imperdiet quis a nulla. In eros libero, consectetur ac nibh feugiat, tristique ultrices nisl. Suspendisse blandit malesuada mauris sed fermentum. Morbi rutrum et nulla quis malesuada. Nulla lacinia arcu eu velit placerat, posuere tristique nibh ultrices. Phasellus consequat leo nisl, nec aliquam velit posuere vitae. Mauris sit amet consequat nunc, in lobortis est. Nam lobortis tincidunt elit sit amet molestie.
    </p>
  </div>
</div>
`;

exports[`testing react components using jest snapshot Testing Quote component 1`] = `
<div
  className="screen"
>
  <div
    className="container"
  >
    <br />
    <br />
    <br />
    <h2>
      Matenatics is not about numbers, equations, computations, or algorithms: is about understanding -William Paul Thurston.
    </h2>
  </div>
</div>
`;
